pyccel/ast/datatypes.py

```diff
@@ -76,8 +76,9 @@
                                  'int32':('int',4),
                                  'int64':('int',8),
                                  'int'  :('int', default_precision['int']),
+                                 'pythonint'  :('int', default_precision['int']),
```

Comment:
Shouldn't `PythonInt` replace `Int` here?

Reply:
I think there are a few `int` s in the code too. We don't want `Int`, but `int` is a little less troublesome.
I tried this on my machine and I had `KeyError`s in the functions `test_array_int_2d_C_initialization`, `test_array_int_2d_F_initialization`, `test_array_real_1d_complex_3d_expr`, `test_array_real_2d_C_complex_3d_expr`, `test_array_real_2d_F_complex_3d_expr` and `test_array_kwargs_full` in `tests/epyccel/test_arrays.py`

Reply:
Actually I have just done a little debugging on this and apparently `int` is used when we pass a `NativeInteger` to `process_dtype`. So either this is a non-issue. Or we should modify `process_dtype` to handle this case separately using `default_precision` instead of `dtype_and_precision_registry`

Reply:
Whichever we choose I will need to do the same for `NativeBool` before the PR can be merged

Reply:
It is also used when the user uses a string to specify the dtype of a numpy array. E.g:
```
tmp = array([[1, 2, 3], [4, 5, 6]], dtype='int', order='F')
```
We therefore need to retain these entries

Reply:
I see, so there seems to be some confusion in the way we use `dtype_and_precision_registry`. Should we meet and discuss about it?

Reply:
The dictionary `dtype_and_precision_registry` is used to set the dtype and precision of a type specified in the grammar (so str->DataType) and in the function `process_dtype` in `ast/numpyext.py` to collect the dtype from the provided argument (so str->DataType (`tmp = array([[1, 2, 3], [4, 5, 6]], dtype='int', order='F')`), CastFunction->DataType (`tmp = array([[1, 2, 3], [4, 5, 6]], dtype=int, order='F')` ) or DataType->DataType (`tmp = array([[1, 2, 3], [4, 5, 6]], order='F')`)
This seems fairly consistent to me and means that it is completely normal to have int  in the list as well as PythonInt